lib/spack/spack/ci/__init__.py

```diff
@@ -413,6 +426,59 @@ def collect_pipeline_options(env: ev.Environment, args) -> PipelineOptions:
     return options
 
 
+def get_unaffected_pruners(
+    env, untouched_pruning_dependent_depth: Optional[int]
+) -> Optional[PrunerCallback]:
+    # Check if the environment changed
+    stack_changed = False
+    rev1, rev2 = get_change_revisions(os.path.dirname(env.manifest_path))
+    if rev1 and rev2:
+        stack_changed = get_stack_changed(env.manifest_path, rev1, rev2)
+
+    # If the stack env has changed, do not apply unaffected pruning
+    if stack_changed:
+        tty.info("Skipping unaffected pruning: stack environment changed")
+        return None
+
+    # TODO: This should be configurable to only check for changed packages
+    # in specific configured repos that are being tested with CI. For now
+    # it assumes all configured repos are merge commits that contain relevant
+    # changes to run CI on.
+    affected_pkgs: List[str] = []
+    for repo in spack.repo.PATH.repos:
+        rev1, rev2 = get_change_revisions(repo.root)
```

Comment:
This is rather odd and hard to follow code. Why not branch on "is this in a git repo yes/no" instead of calling a function to optionally return a static tupel of strings.

Reply:
so yes, the `get_change_revisions` is currently doing a hard coded thing. It assumes that you are always inspecting a merge commit which may not necessarily be correct for all cases, but it is for our main case. The function exists as a long term placeholder so we can document that limitation in a single place and make clear the assumptions being used when it is called.